test/sqllogictest/scalar_subqueries_select_list.slt

```diff
@@ -801,23 +801,33 @@ Explained Query:
         ArrangeBy keys=[[#0, #1]] // { arity: 3 }
           Union // { arity: 3 }
             Get l3 // { arity: 3 }
-            Map (null) // { arity: 3 }
-              Union // { arity: 2 }
-                Negate // { arity: 2 }
-                  Distinct project=[#0, #1] // { arity: 2 }
-                    Project (#0, #1) // { arity: 2 }
-                      Get l3 // { arity: 3 }
-                Get l1 // { arity: 2 }
+            Project (#0, #1, #4) // { arity: 3 }
```

Comment:
This looks like an interesting regression! No idea, but something about optimizing reduce expressions results in a worse plan! Looks a bit like semijoin elimination (`l1` is distinct, and .. idk perhaps how redundant join reasons about column provenance breaks if we are too smart and replace them with equivalent columns.